src/tool/xmeta/cpp/models/class_or_interface_model.h

```diff
@@ -34,46 +43,36 @@ namespace xlang::xmeta
             return m_events;
         }
 
-        void add_interface_base_ref(std::string_view const& interface_base_ref)
-        {
-            m_interface_base_refs.emplace_back(std::string(interface_base_ref));
-        }
+        void add_interface_base_ref(std::string_view const& interface_base_ref);
 
-        void add_interface_base_ref(size_t index, std::shared_ptr<interface_model> interface_base_ref)
-        {
-            assert(index < m_interface_base_refs.size());
-            m_interface_base_refs[index].resolve(interface_base_ref);
-        }
+        compilation_error add_member(std::shared_ptr<method_model> const& member);
+        compilation_error add_member(std::shared_ptr<property_model> const& member);
+        compilation_error add_member(std::shared_ptr<event_model> const& member);
 
-        void add_member(std::shared_ptr<property_model> const& member)
-        {
-            m_properties.emplace_back(member);
-        }
+        std::shared_ptr<method_model> const& get_method_member(std::string const& member_id);
+        std::shared_ptr<property_model> const& get_property_member(std::string const& member_id);
 
-        void add_member(std::shared_ptr<method_model> const& member)
-        {
-            m_methods.emplace_back(member);
-        }
 
-        void add_member(std::shared_ptr<event_model> const& member)
-        {
-            m_events.emplace_back(member);
-        }
+        bool member_id_exists(std::string_view const& id);
+        bool event_or_property_id_exists(std::string_view const& id);
```

Comment:
standardize on "name" instead of "id" across the tool

Reply:
Note, you can have this method signature be "bool event_or_property_exists(std::string_view const& name)"